### PR TITLE
codebase/introduction-to-automq [BAEL-8707]

### PR DESCRIPTION
**Pull request by hardikSinghBehl**
Codebase for the article: **Introduction to AutoMQ**

### CODE REVIEW

File: messaging-modules/automq/src/main/java/com/baeldung/automq/UserOnboardingInitiatedListener.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.kafka.annotation.KafkaListener;
+
+@Configuration
+public class UserOnboardingInitiatedListener {
+
+    private static final Logger log = LoggerFactory.getLogger(UserOnboardingInitiatedListener.class);
+
+    @KafkaListener(topics = { "${com.baeldung.topic.onboarding-initiated}" }, groupId = "user-service")
```
Comment (theangrydev) [inline messaging-modules/automq/src/main/java/com/baeldung/automq/UserOnboardingInitiatedListener.java:13]: Is it possible to use the shorthand for a single element array here?
```suggestion
    @KafkaListener(topics = "${com.baeldung.topic.onboarding-initiated}", groupId = "user-service")
```

**Review by theangrydev — CHANGES_REQUESTED** [review]
I see this when trying to run the build locally:
```
Caused by: java.lang.IllegalStateException: java.lang.NoSuchMethodError: 'java.util.concurrent.locks.ReentrantLock ch.qos.logback.classic.LoggerContext.getConfigurationLock()'
```

File: messaging-modules/automq/src/main/java/com/baeldung/automq/UserOnboardingInitiatedListener.java (hunk 1 above)
Comment (hardikSinghBehl) [reply messaging-modules/automq/src/main/java/com/baeldung/automq/UserOnboardingInitiatedListener.java:13]: Yes, review incorporated.

**Comment by hardikSinghBehl** [comment]
> I see this when trying to run the build locally:
> 
> ```
> Caused by: java.lang.IllegalStateException: java.lang.NoSuchMethodError: 'java.util.concurrent.locks.ReentrantLock ch.qos.logback.classic.LoggerContext.getConfigurationLock()'
> ```

Added `logback-spring.xml` to resolve the above issue. 
The test is running successfully now: `mvn test -Dtest=UserOnboardingInitiatedListenerLiveTest`

**Pull request merged**